pkg/vm/common.go

```
@@ -17,19 +19,45 @@ func New(b *builder.Build) *Script {
 	}
 }
 
-func (s *Script) Run() (*v8go.Value, func(), error) {
-	ctx := v8go.NewContext()
+func (s *Script) Run(payload []byte) (*v8.Value, func(), error) {
+	iso := v8.NewIsolate()
+	ctx := v8.NewContext(iso)
 
 	// Compile the script.
-	script, err := ctx.Isolate().CompileUnboundScript(s.code, s.path, v8go.CompileOptions{CachedData: s.cache})
+	script, err := iso.CompileUnboundScript(s.code, s.path, v8.CompileOptions{CachedData: s.cache})
 	if err != nil {
-		return nil, ctx.Close, err
+		return nil, iso.Dispose, err
 	}
 
+	logFn := v8.NewFunctionTemplate(iso, func(info *v8.FunctionCallbackInfo) *v8.Value {
+		if args := info.Args(); len(args) > 0 {
+			for _, input := range args {
+				message.Info(input.String())
+			}
+		}
+
+		return nil
+	})
+
+	console := v8.NewObjectTemplate(iso)
+	console.Set("log", logFn, v8.ReadOnly)
+
+	consoleObj, _ := console.NewInstance(ctx)
+
+	global := ctx.Global()
+
+	global.Set("console", consoleObj)
+	global.Set("__pepr_payload", string(payload))
+
 	// Run the script.
-	out, err := script.Run(ctx)
+	_, err = script.Run(ctx)
+	if err != nil {
+		return nil, iso.Dispose, err
+	}
+
+	out, err := ctx.RunScript("global.transform(__pepr_payload)", "main.js")
 	if err != nil {
-		return nil, ctx.Close, err
+		fmt.Println(err)
```

assume this was for debugging purposes, but we should return the error/close handler vs just log it

---

oh yes, and also this should do the iso.Dispose as well.

---

if we get to reusable isolates per-script (1:N script to instantiations on isolate vs. context) top level failures should be iso.Dispose vs. ctx.Close